cmd/utils/flags.go

```diff
@@ -29,84 +49,99 @@ func AddCommonFlags(command *cobra.Command, flags *CommonFlags) {
 // (1) Provide the kubernetes node name
 // (2) Provide the VMMS instance information (--subscription, --node-resource-group, --vmss and --instance-id)
 // (3) Provide Resource ID (/subscriptions/mySubID/resourceGroups/myRG/providers/myProvider/virtualMachineScaleSets/myVMSS/virtualMachines/myInsID)
-func AddNodeFlags(command *cobra.Command, vm *VirtualMachineScaleSetVM) {
-	var (
-		node              string
-		subscriptionID    string
-		nodeResourceGroup string
-		vmScaleSet        string
-		instanceID        string
-		resourceID        string
-	)
-
+func AddNodeFlags(command *cobra.Command, useFlagsOnly bool) {
 	command.PersistentFlags().StringVarP(
 		&node,
-		"node", "",
+		NodeKey, "",
 		"",
 		"Kubernetes node name.",
 	)
 	command.PersistentFlags().StringVarP(
 		&subscriptionID,
-		"subscription", "",
+		SubscriptionIDKey, "",
 		"",
 		"Subscription ID.",
 	)
 	command.PersistentFlags().StringVarP(
 		&nodeResourceGroup,
-		"node-resource-group", "",
+		NodeResourceGroupKey, "",
 		"",
 		"Node resource group name.",
 	)
 	command.PersistentFlags().StringVarP(
-		&vmScaleSet,
-		"vmss", "",
+		&vmss,
+		VMSSKey, "",
 		"",
 		"Virtual machine scale set name.",
 	)
 	command.PersistentFlags().StringVarP(
-		&instanceID,
-		"instance-id", "",
+		&vmssInstanceID,
+		VMSSInstanceIDKey, "",
 		"",
 		"VM scale set instance ID.",
 	)
 	command.PersistentFlags().StringVarP(
 		&resourceID,
-		"id", "",
+		ResourceIDKey, "",
 		"",
 		`Resource ID containing all information of the VMSS instance using format:
 		e.g. /subscriptions/mySubID/resourceGroups/myRG/providers/myProvider/virtualMachineScaleSets/myVMSS/virtualMachines/myInsID.
 		Notice it is not case sensitive.`,
 	)
 
 	command.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
-		if node != "" {
-			if resourceID != "" {
-				return errors.New("specify either --node or --id but not both")
+		if !useFlagsOnly {
+			if err := loadCurrentNodeConfig(); err != nil {
+				return fmt.Errorf("loading config: %w", err)
 			}
+			// bind environment variables
+			config.AutomaticEnv()
+			config.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
```

Comment:
@blanquicet should we set a prefix for the environment variables? We can do this by `config.SetEnvPrefix("kubectl_az")` the reason I ask is because `NODE` seems bit too generic so maybe `KUBECTL_AZ_NODE` can be a better option. What do you say?

Reply:
I think that's a great idea. I'm just worried about `nodeResourceGroup` as it would be `KUBECTL_AZ_NODE_RESOURCE_GROUP`

Reply:
Since it follows a pattern I think we are fine there, otherwise the envs were too generic!